pkg/reconciler/v1alpha1/pipelinerun/resources/conditionresolution_test.go

```diff
@@ -0,0 +1,288 @@
+			},
+			Spec: v1alpha1.TaskRunSpec{},
+			Status: v1alpha1.TaskRunStatus{
+				Status: duckv1beta1.Status{
+					Conditions: []apis.Condition{{
+						Type:   apis.ConditionSucceeded,
+						Status: corev1.ConditionTrue,
+					}},
+				},
+			},
+		},
+	},
+}
+
+var failedState = TaskConditionCheckState{
+	{
+		ConditionCheckName: "foo",
+		Condition:          c,
+		ConditionCheck: &v1alpha1.ConditionCheck{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "failed-condition-check",
+			},
+			Spec: v1alpha1.TaskRunSpec{},
+			Status: v1alpha1.TaskRunStatus{
+				Status: duckv1beta1.Status{
+					Conditions: []apis.Condition{{
+						Type:   apis.ConditionSucceeded,
+						Status: corev1.ConditionFalse,
+					}},
+				},
+			},
+		},
+	},
+}
+
+func TestTaskConditionCheckState_HasStarted(t *testing.T) {
+	tcs := []struct {
+		name  string
+		state TaskConditionCheckState
+		want  bool
+	}{
+		{
+			name:  "no-condition-checks",
+			state: notStartedState,
+			want:  false,
+		},
+		{
+			name:  "running-condition-check",
+			state: runningState,
+			want:  true,
+		},
+		{
+			name:  "successful-condition-check",
+			state: successState,
+			want:  true,
+		},
+		{
+			name:  "failed-condition-check",
+			state: failedState,
+			want:  true,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			got := tc.state.HasStarted()
+			if d := cmp.Diff(got, tc.want); d != "" {
+				t.Errorf("Expected HasStarted to be %v but got %v for %s", tc.want, got, tc.name)
+			}
+		})
+	}
+}
+
+func TestTaskConditionCheckState_IsComplete(t *testing.T) {
+	tcs := []struct {
+		name  string
+		state TaskConditionCheckState
+		want  bool
+	}{
+		{
+			name:  "no-condition-checks",
+			state: notStartedState,
+			want:  false,
+		},
+		{
+			name:  "running-condition-check",
+			state: runningState,
+			want:  false,
+		},
+		{
+			name:  "successful-condition-check",
+			state: successState,
+			want:  true,
+		},
+		{
+			name:  "failed-condition-check",
+			state: failedState,
+			want:  true,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			got := tc.state.IsComplete()
+			if d := cmp.Diff(got, tc.want); d != "" {
+				t.Errorf("Expected IsComplete to be %v but got %v for %s", tc.want, got, tc.name)
+			}
+		})
+	}
+}
+
+func TestTaskConditionCheckState_IsSuccess(t *testing.T) {
+	tcs := []struct {
+		name  string
+		state TaskConditionCheckState
+		want  bool
+	}{
+		{
+			name:  "no-condition-checks",
+			state: notStartedState,
+			want:  false,
+		},
+		{
+			name:  "running-condition-check",
+			state: runningState,
+			want:  false,
+		},
+		{
+			name:  "successful-condition-check",
+			state: successState,
+			want:  true,
+		},
+		{
+			name:  "failed-condition-check",
+			state: failedState,
+			want:  false,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			got := tc.state.IsSuccess()
+			if d := cmp.Diff(got, tc.want); d != "" {
```

Comment:
hm i think it might be simpler to compare `got` and `tc.want` directly since they are both bools?

Reply:
:man_facepalming: